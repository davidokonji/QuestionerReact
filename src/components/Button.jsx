import React from 'react';

const Button = (props) => {
  const { text, event, styles } = props;
  return (
    <button type="button" onClick={event} className={styles}>
      {text}
    </button>
  );
};

export default Button;
