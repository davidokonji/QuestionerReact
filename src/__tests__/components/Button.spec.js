import React from 'react';
import { shallow } from 'enzyme';
import Button from '../../components/Button';

describe('Button component', () => {
  test('should render button component correctly', () => {
    const wrapper = shallow(<Button />);

    expect(wrapper).toMatchSnapshot();
  });

  test('should render button component with props', () => {
    const wrapper = shallow(<Button text="hello" styles="hellostyle" />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.hasClass('hellostyle')).toBeTruthy();
  });
});
