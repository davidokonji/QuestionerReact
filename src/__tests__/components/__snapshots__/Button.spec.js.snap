// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render button component correctly 1`] = `
<button
  type="button"
/>
`;

exports[`Button component should render button component with props 1`] = `
<button
  className="hellostyle"
  type="button"
>
  hello
</button>
`;
